core/templates/domain/classroom/classroom-backend-api.service.spec.ts

```diff
@@ -202,4 +202,260 @@ describe('Classroom backend API service', function() {
       expect(failHandler).toHaveBeenCalled();
     })
   );
+
+  it('should create new classroom id', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let service = classroomBackendApiService;
+
+    service.getNewClassroomIdAsync().then(successHandler, failHandler);
+    let req = httpTestingController.expectOne(
+      '/classroom_id_handler');
+    expect(req.request.method).toEqual('GET');
+    req.flush({classroom_id: 'classroomId'});
+
+    flushMicrotasks();
+    expect(successHandler).toHaveBeenCalledWith('classroomId');
```

Comment:
```suggestion
    flushMicrotasks();

    expect(successHandler).toHaveBeenCalledWith('classroomId');
```
Please make similar changes everywhere

Reply:
Updated